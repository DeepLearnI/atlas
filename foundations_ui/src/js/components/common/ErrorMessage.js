import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorMessage extends Component {
  constructor(props) {
    super(props);
    this.setContent = this.setContent.bind(this);
    this.setInternalServerError = this.setInternalServerError.bind(this);
    this.setNotFoundError = this.setNotFoundError.bind(this);
    this.setBadRequestError = this.setBadRequestError.bind(this);
    this.setUnauthorizedError = this.setUnauthorizedError.bind(this);
    this.state = {
      errorCode: this.props.errorCode,
    };
  }

  setContent() {
    const { errorCode } = this.state;
    if (errorCode === 404) {
      return this.setNotFoundError();
    }
    if (errorCode === 500) {
      return this.setInternalServerError();
    }
    if (errorCode === 400) {
      return this.setBadRequestError();
    }
    if (errorCode === 401) {
      return this.setUnauthorizedError();
    }

    return {
      errorBanner: '',
      errorSubtext: '',
    };
  }

  setInternalServerError() {
    const errorBanner500 = '500 Internal Server Error';
    const errorSubtext500 = 'Our servers are having problems '
    + 'going through the astroid belts.'
    + ' Check back again shortly.';
    return {
      errorBanner: errorBanner500,
      errorSubtext: errorSubtext500,
    };
  }

  setNotFoundError() {
    const errorBanner404 = '404 Page Not Found';
    const errorSubtext404 = 'Services are having temporary issues.'
    + ' Contact our front desk support at support@dessa.com or call us toll free at 1-899-623-5578.';
    return {
      errorBanner: errorBanner404,
      errorSubtext: errorSubtext404,
    };
  }

  setBadRequestError() {
    const errorBanner400 = '400 Bad Request Error';
    const errorSubtext400 = 'The request was malformed, please try a proper request.';
    return {
      errorBanner: errorBanner400,
      errorSubtext: errorSubtext400,
    };
  }

  setUnauthorizedError() {
    const errorBanner401 = '401 Unauthorized';
    const errorSubtext401 = <a href="/login">Please Login here</a>;
    return {
      errorBanner: errorBanner401,
      errorSubtext: errorSubtext401,
    };
  }


  render() {
    const { errorBanner, errorSubtext } = this.setContent();

    return (
      <div className="error-body-container">
        <div className="i--icon-astronaut-probs text-center" />
        <h1 className="blue-border-bottom font-bold">Houston, we have a problem.</h1>
        <h2 className="font-bold">{errorBanner}</h2>
        <p>{errorSubtext}</p>
      </div>
    );
  }
}

ErrorMessage.propTypes = {
  errorCode: PropTypes.number,
};

ErrorMessage.defaultProps = {
  errorCode: 404,
};

export default ErrorMessage;
