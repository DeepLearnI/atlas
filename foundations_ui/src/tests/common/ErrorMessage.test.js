import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';
import ErrorMessage from '../../js/components/common/ErrorMessage';

configureTests();

it('Shallow Renders ErrorMessage', () => {
  const wrapper = shallow(<ErrorMessage/>);
  expect(wrapper).toMatchSnapshot();
});

it('Calls Correct Function for 500 Error', () => {
  const spy_content = jest.spyOn(ErrorMessage.prototype, 'setContent');
  const spy_500 = jest.spyOn(ErrorMessage.prototype, 'setInternalServerError');
  const wrapper = shallow(<ErrorMessage errorCode={500}/>);
  expect(spy_content).toHaveBeenCalled();
  expect(spy_500).toHaveBeenCalled();
});

it('Returns Correct Banner and Subtext for 500 Error', () => {
  const wrapper = shallow(<ErrorMessage errorCode={500}/>);
  const returnValues = wrapper.instance().setContent();
  expect(returnValues.errorBanner).toEqual('500 Internal Server Error');
  expect(returnValues.errorSubtext).toEqual('Our servers are having problems '
  + 'going through the astroid belts.'
  + ' Check back again shortly.')
});

it('Calls Correct Function for 404 Error', () => {
  const spy_content = jest.spyOn(ErrorMessage.prototype, 'setContent');
  const spy_404 = jest.spyOn(ErrorMessage.prototype, 'setNotFoundError');
  const wrapper = shallow(<ErrorMessage errorCode={404}/>);
  expect(spy_content).toHaveBeenCalled();
  expect(spy_404).toHaveBeenCalled();
});

it('Returns Correct Banner and Subtext for 404 Error', () => {
  const wrapper = shallow(<ErrorMessage errorCode={404}/>);
  const returnValues = wrapper.instance().setContent();
  expect(returnValues.errorBanner).toEqual('404 Page Not Found');
  expect(returnValues.errorSubtext).toEqual('Services are having temporary issues.'
  + ' Contact our front desk support at support@dessa.com or call us toll free at 1-899-623-5578.')
});