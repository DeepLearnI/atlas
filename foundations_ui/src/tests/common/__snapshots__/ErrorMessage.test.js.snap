// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ErrorMessage 1`] = `
<div
  className="error-body-container"
>
  <div
    className="i--icon-astronaut-probs text-center"
  />
  <h1
    className="blue-border-bottom font-bold"
  >
    Houston, we have a problem.
  </h1>
  <h2
    className="font-bold"
  >
    404 Page Not Found
  </h2>
  <p>
    Services are having temporary issues. Contact our front desk support at support@dessa.com or call us toll free at 1-899-623-5578.
  </p>
</div>
`;
